tests/test_commands.py

```diff
@@ -0,0 +1,188 @@
+import os
+import signal
+import shutil
+import subprocess
+import sys
+import time
+
+import pytest
+import requests
+import sh
+
+
+FIXTURES = 'tests/fixtures'
+WEBSITE = os.path.join(FIXTURES, 'website.py')
+BUILDDIR = os.path.join(FIXTURES, '_build')
+INDEX = os.path.join(BUILDDIR, 'index.html')
+CNAME = os.path.join(BUILDDIR, 'CNAME')
+
+
+def parametrized_fixture(*args, **kwargs):
+    @pytest.fixture(scope='module', params=args + tuple(kwargs.keys()))
+    def fixture(request):
+        if request.param in kwargs:
+            return kwargs[request.param]
+        return request.param
+    return fixture
+
+
+class Cname(str):
+    '''
+    Helper class that is str, but has different truthfulness
+    '''
+    def __bool__(self):
+        return self == ' --cname'
+
+
+cname = parametrized_fixture(cname=Cname(' --cname'),
+                             no_cname=Cname(' --no-cname'))
+serve_command = parametrized_fixture('serve', freeze_serve='freeze --serve')
+port = parametrized_fixture(8001, 8080)
+domain = parametrized_fixture('foo.bar', 'spam.eggs')
+protocol = parametrized_fixture('http', 'https')
+
+
+class ElsaRunner:
+    '''
+    Class for elsa fixture enabling multiple runs and exit status check
+    '''
+    def __init__(self):
+        self.runs = []
+
+    def run(self, command):
+        print('COMMAND: python website.py', command)
+        proc = subprocess.Popen([sys.executable, WEBSITE] + command.split())
+        if 'serve' in command:
+            time.sleep(1)
+        else:
+            proc.communicate()
+        self.runs.append((proc, command))
+
+    def finalize(self):
+        try:
+            for proc, command in self.runs:
+                proc.terminate()
+                if 'serve' in command:
+                    proc.communicate()
+                assert (proc.returncode == 0 or
+                        proc.returncode == -signal.SIGTERM)
+        finally:
+            self.lax_rmtree(BUILDDIR)
+            self.lax_rmbranch('gh-pages')
+
+    @classmethod
+    def lax_rmtree(cls, path):
+        try:
+            shutil.rmtree(path)
+        except FileNotFoundError:
+            pass
+
+    @classmethod
+    def lax_rmbranch(cls, branch):
+        try:
+            sh.git.branch('-D', branch)
+        except sh.ErrorReturnCode_1:
+            pass
+
+
+@pytest.fixture
+def elsa():
+    er = ElsaRunner()
+    try:
+        yield er.run
+    finally:
+        er.finalize()
```

Comment:
I think it would work better if `ElsaRunner.run` was a context manager. Here `ElsaRunner.finalize` has an assert, and it's better to have that as part of the test rather in a fixture finalizer.

Removing BUILDDIR and gh-pages, on the other hand, is a perfect fit for fixtures.